src/types.jl

```diff
@@ -414,8 +414,16 @@ AGray32(g::AbstractGray, alpha = 1) = AGray32(gray(g), alpha)
 # Note: with the exceptions of `alphacolor` and `coloralpha`, all
 # traits in the rest of this file are intended just for internal use
 
-const color3types = filter(x->(!x.abstract && length(fieldnames(x))>1), union(subtypes(Color), subtypes(AbstractRGB)))
-const parametric3 = filter(x->!isempty(x.parameters), color3types)
+const color3types = map(s->getfield(ColorTypes,s),
+  filter(names(ColorTypes, true)) do s
+    isdefined(ColorTypes, s) || return false
+    t = getfield(ColorTypes, s)
+    isa(t, Type) && t <: Colorant && !isabstract(t) && length(fieldnames(t))>1
```

Comment:
julia 0.5:
```julia
julia> using Compat.TypeUtils

julia> using ColorTypes

julia> typealias Color1{T} Color{T,1}
ColorTypes.Color{T,1}

julia> isabstract(Color1)
ERROR: MethodError: no method matching isabstract(::Type{ColorTypes.Color{T,1}})
Closest candidates are:
  isabstract(::DataType) at /home/tim/.julia/v0.5/Compat/src/Compat.jl:1781
```
This causes a precompilation error. Fixed if we add `isabstract(::TypeConstructor) = true` to Compat.